### ISSUE TITLE
CLJS compatibilty

### ISSUE THREAD

**Issue by l3nz** [l3nz/cli-matic#19]
Libs are supported.... what is to be done?

**Comment by arichiardi** [comment]
This is a good idea, I would be interested in trying this lib out and eventually replace `tools.cli`.

Checking the code...

**Comment by l3nz** [comment]
I expect that it should be trivial; it is a matter of implementing namespaces `platform.cljs` (where JVM stuff is) and `optional.cljs` where we have optional stuff - the only "optional" that I guess will be there initially is the ability to decode JSON that JS surely has.

Please note that I created the current cljs namespaces in a hurry and functions there just return nil, all of them - so they will break sth else down the line.

**Comment by l3nz** [comment]
Link https://medium.com/@jamesleonis/step-inside-cljsbuild-e38271b10415

**Comment by arichiardi** [comment]
Thank you for the quick answer! Maybe the link is not really the one you wanted to post? I see it points to a leiningen post :smile:

**Comment by arichiardi** [comment]
Oh I see, you wanted to write down the doc to how to configure lein

**Comment by l3nz** [comment]
Now I have a clean compile (kind of) but tests still do not pass. Want to give it a try?

**Comment by arichiardi** [comment]
Yep will give it a try, no promise about time but I will do my best :smile:

**Comment by l3nz** [comment]
I'm starting to think we should target Planck initially, as it seems the most likely thing one would use to write a CLI app.

**Comment by arichiardi** [comment]
The classic approach is to implement ClojureScript, keeping the amount of macros very low and then port it to self-host. I have done a bunch of those so I can help with that.

**Comment by l3nz** [comment]
Link to keep track of https://github.com/abiocljs/abio

**Comment by l3nz** [comment]
We have an initial working system with Planck: https://github.com/l3nz/cli-matic/blob/master/planck.md

**Comment by arichiardi** [comment]
Thank you! Not a plank user (yet) but I see no reason why it shouldn't work on any self-host environment (like lumo)

**Comment by l3nz** [comment]
Yes, but you have to start somewhere :) at this point, as it's working,  we cn just change the platforms.cljs adding stuff there for various environments! :smile:
